### PR TITLE
[READY] dilutes the staff of chaos pool with some more bolts

### PR DESCRIPTION
**Pull request by tralezab**
Now that fugitives are mostly finished and have entered testmerge phase, I can finally drop this bad boy!

:cl:
add: The wizard has many more staves to play with!
add: Along with the new effects, the chaos bolt has a couple more bolts not available my normal means
/:cl:
Why?
Staff of chaos is actually not that chaotic, it's pretty boring. Why don't we add some more effects to change it from "staff of change but it takes longer" to "oh god oh fuck"

And we'll do it with these staves:
Er- i'm redoing this. My first draft (check edits!) had too many positive or weaker effects, making the chaos staff not good for wizards.

### CODE REVIEW

**Comment by tralezab** [comment]
managed to half fuck my git trying to do a cool squash trick but oh well

**Comment by Jordie0608** [comment]
Don't worry about squashing; it's very simple to do when merging and most of the time just makes reviewing differences more tedious.

**Comment by Cyprex** [comment]
It would be cool to have the staff of awakening and staff of the law as adminspawn only.
> Staff of grounding: Any flying mobs take massive damage and have their flying removed
This seems incredibly niche, to the point of being useless.

File: code/modules/projectiles/guns/magic/staff.dm [hunk 1]
```diff
@@ -105,4 +105,69 @@
 	max_charges = 6
 	recharge_rate = 4
 
+/obj/item/gun/magic/staff/flying
+	name = "staff of flying"
+	desc = "An artefact that spits bolts of graceful magic that can make something fly."
```
Comment (Borisvanmemes) [inline code/modules/projectiles/guns/magic/staff.dm]: Typo, artifact is spelled with a I, not a E.

File: code/modules/projectiles/guns/magic/staff.dm [hunk 2]
```diff
@@ -105,4 +105,69 @@
...
+	fire_sound = 'sound/magic/staff_change.ogg'
+	ammo_type = /obj/item/ammo_casing/magic/flying
+	icon_state = "staffofchange"
+	item_state = "staffofchange"
+
+/obj/item/gun/magic/staff/grounding
+	name = "staff of grounding"
+	desc = "An artefact that spits bolts of dense magic that hates flying things."
```
Comment (Borisvanmemes) [inline code/modules/projectiles/guns/magic/staff.dm]: see above

File: code/modules/projectiles/guns/magic/staff.dm [hunk 3]
```diff
@@ -105,4 +105,69 @@
...
+	fire_sound = 'sound/magic/staff_change.ogg'
+	ammo_type = /obj/item/ammo_casing/magic/grounding
+	icon_state = "staffofchange"
+	item_state = "staffofchange"
+
+/obj/item/gun/magic/staff/bounty
+	name = "staff of bounty"
+	desc = "An artefact that spits bolts of latent magic that can provide some delayed rewards to the bloodshed."
```
Comment (Borisvanmemes) [inline code/modules/projectiles/guns/magic/staff.dm]: and here.

File: code/modules/projectiles/guns/magic/staff.dm [hunk 4]
```diff
@@ -105,4 +105,69 @@
...
+	fire_sound = 'sound/magic/staff_change.ogg'
+	ammo_type = /obj/item/ammo_casing/magic/bounty
+	icon_state = "staffofchange"
+	item_state = "staffofchange"
+
+/obj/item/gun/magic/staff/antimagic
+	name = "staff of antimagic"
+	desc = "An artefact that spits bolts of paradoxical magic that magically prevents magic."
```
Comment (Borisvanmemes) [inline code/modules/projectiles/guns/magic/staff.dm]: and also here.

File: code/modules/projectiles/guns/magic/staff.dm [hunk 5]
```diff
@@ -105,4 +105,69 @@
...
+	fire_sound = 'sound/magic/staff_change.ogg'
+	ammo_type = /obj/item/ammo_casing/magic/antimagic
+	icon_state = "staffofchange"
+	item_state = "staffofchange"
+
+/obj/item/gun/magic/staff/sapping
+	name = "staff of sapping"
+	desc = "An artefact that spits bolts of depressed magic that makes you feel pretty bad."
```
Comment (Borisvanmemes) [inline code/modules/projectiles/guns/magic/staff.dm]: annnnd here.

File: code/modules/projectiles/guns/magic/staff.dm [hunk 6]
```diff
@@ -105,4 +105,69 @@
...
+	ammo_type = /obj/item/ammo_casing/magic/sapping
+	icon_state = "staffofchange"
+	item_state = "staffofchange"
+
+
+//ADMIN ONLY FROM HERE ON OUT//
+
+/obj/item/gun/magic/staff/law
+	name = "staff of the law"
+	desc = "An artefact that spits bolts of vindicative magic that detains criminal scum."
```
Comment (Borisvanmemes) [inline code/modules/projectiles/guns/magic/staff.dm]: this too.

File: code/modules/projectiles/guns/magic/staff.dm [hunk 7]
```diff
@@ -105,4 +105,69 @@
...
+	fire_sound = 'sound/magic/staff_change.ogg'
+	ammo_type = /obj/item/ammo_casing/magic/law
+	icon_state = "staffofchange"
+	item_state = "staffofchange"
+
+/obj/item/gun/magic/staff/awakening
+	name = "staff of awakening"
+	desc = "An artefact that spits bolts of unlocking magic that can make someone realize their inner powers."
```
Comment (Borisvanmemes) [inline code/modules/projectiles/guns/magic/staff.dm]: almost done.

File: code/modules/projectiles/guns/magic/staff.dm [hunk 8]
```diff
@@ -105,4 +105,69 @@
...
+	fire_sound = 'sound/magic/staff_change.ogg'
+	ammo_type = /obj/item/ammo_casing/magic/awakening
+	icon_state = "staffofchange"
+	item_state = "staffofchange"
 
+/obj/item/gun/magic/staff/gib
+	name = "staff of gravitokinetics"
+	desc = "An artefact that spits bolts of active magic that explodes victims."
```
Comment (Borisvanmemes) [inline code/modules/projectiles/guns/magic/staff.dm]: annnd this is the last one,

**Review by Borisvanmemes — CHANGES_REQUESTED** [review]
Typos, pls fix.

**Comment by YakumoChen** [comment]
> 
> 
> Typos, pls fix.

Wrong, there are two spellings, and the code already has multiple instances of the spelling used in the commit.

**Comment by Borisvanmemes** [comment]
i am a fool and a sham.

**Comment by OverlyGeneric** [comment]
There are several problems with the staves with one being redundant against it's intended counter, one being too beneficial for the crew, etcetera.

I would recommend giving the staves several thematic variants to their bolts; acting like lesser variants of the staff of chaos.

The staff of flying should have three effects like granting temporary flight instead of permanent, flinging the victim in a random direction, and polymorphing the victim into a chicken for a minute.

The staff of grounding is too niche, give it three effects such as dropping the victim to the ground like gravity shifted, warping them to the brig's jail cells, and freezing the victim in place before shocking them with all the electricity in the room.

The staff of bounty seems a bit powerful but overall fine with its current bolt. I would recommend adding two other bolts though like creating food around the victim and causing the "marked for death" effect from Bubblegum's loot to the victim. 

The staff of sapping should have some variants like causing stamina damage overtime and draining charge from any battery-charged devices.

**Reply if you need any other ideas. I have plenty.**

**Comment by Shadowflame909** [comment]
how does staff of flight work? Is it like angel wings?

File: code/modules/projectiles/guns/magic/staff.dm (hunk 2 above)
Comment (ATH1909) [reply code/modules/projectiles/guns/magic/staff.dm]: Remove the "s" in "hates", since that presumably refers to the bolts, not the staff.

**Comment by ATH1909** [comment]
You should add some of these staves to the summon magic list.

**Comment by tralezab** [comment]
I had this near finished but a surprise winter storm hit sorry fellas

**Comment by tralezab** [comment]
@Shadowflame909 it sets flying, and if flying is removed it grants flying again if the status effect is still on you.

@Cyprex combos with the flight staff to repeatedly smash someone into the ground. Use it well to deal massive damage!

**Comment by ATH1909** [comment]
> @Shadowflame909 it sets flying, and if flying is removed it grants flying again if the status effect is still on you.
> 
> @Cyprex combos with the flight staff to repeatedly smash someone into the ground. Use it well to deal massive damage!

Is there any way to get flight gained from your race back if you lose it to a grounding bolt? Would you just have to toggle a jetpack on and off again or something?

Making a grounding bolt make its target immune to electricity for a minute would be a pretty neat feature.

Maybe make a bolt of flight make stuff (including objects) hit by it get tossed backwards as well (maybe only if they can already fly?)?

**Comment by tralezab** [comment]
@ATH1909

Probably not if you are a simplemob, you'll just be stuck without flight. I'll add the Todd effect for flight so it can be a little more aggressive

**Comment by PKPenguin321** [comment]
Overall a nerf to the staff of chaos, not really a fan of that

**Comment by OverlyGeneric** [comment]
>combos with the flight staff to repeatedly smash someone into the ground. Use it well to deal massive damage!

It's too niche then. Your opponents would be zipping around the map like a madman and you expect someone to manage to NOT disarm you while your chain-stunning?

**Comment by tralezab** [comment]
I can make it cheaper, but I really hoped to keep the combo. Any ideas? I saw the above for grounding and it seems okay

@PKPenguin321 I wanted the effects to be more diverse, but leave the staff at the power level it is now. I may add another deadly staff to replace flight + grounding even if I like the combination of them

**Comment by PKPenguin321** [comment]
if you're running staff of chaos:
*new negatives:*
- law bolt: dud, just a worse locker bolt for no good reason
- flying bolt: dud, just helps the crew
- grounding bolt: dud, unless they were previously hit by a flying bolt (which would be rare)
- awakening bolt: huge negative, can potentially give crew spells to instagib you with
- antimagic bolt: huge negative, makes whoever is hit with it effectively invincible to you (antimagic FUCKS wizards harder than almost anything else)
*new positives*
- gib bolt: negligible positive since death bolt is already a thing? only notably better if you summoned magic and the crew has healing staffs
- sapping bolt: positive, though i'm not sure what it actually does/if it's potent enough to be comparable to fireball

tl;dr you basically doubled the chance that the chaos staff will backfire with next to nothing to compensate, and made some of the backfires absolutely crippling (awakening, antimagic) for the wizard. chaos staff will be made from most fun staff to useless

File: code/modules/projectiles/guns/magic/staff.dm (hunk 2 above)
Comment (tralezab) [reply code/modules/projectiles/guns/magic/staff.dm]: need to do the second thing

**Comment by tralezab** [comment]
Okay, I'll see how I can redo this with staves that are deadly, but in different ways instead of anomalous in different ways

**Comment by YakumoChen** [comment]
awakening makes it so chaos magicarp can randomly give people spells too

**Comment by tralezab** [comment]
I hate those little shits

**Comment by tralezab** [comment]
Okay, here's a couple ideas on some more "potent" spells to dissuade using it as the staff of change slot machine.

* Staff of flight now throws you like mjolnir instead
* Staff of grounding is gone, you will be missed by nobody but me for your ability to play human basketball
* antimagic bolt may get removed but i need to mull over it more, i think with a lot of bonuses this is alright but we can't have BOTH antimagic and awakening. even so it needs a nerf in duration
* awakening see above
* staff of the law is replaced by staff of necropotence, which is soul tap in bolt form (removes max health to refresh spells, usually a good effect unless the target has spells as well as you)
* might remove gib just because it's not interesting

this should make chaos more consistently dangerous but still have a few duds to balance it out. gonna keep spriting and all that

**Comment by ATH1909** [comment]
Make a staff of illusions that summons an illusory copy of the person whom you hit with it to attack them (it'd have stats based on the stats of the struck target); that'd be cool and a neat positive staff of change effect.

Maybe add summoning ash storms in rooms on the station (like that staff that you can get from the legion megafauna) to the list of possible chaos staff effects (with the chaos staff making you immune to said ash storms, because constantly taking burn damage from your own ash storms is lame)?

You could take my suggestion of making the staff of grounding able to toggle between "grounding mode" and "flight/knockback mode", combining the two staves into one staff (don't let staves of chaos shoot grounding bolts, though).

**Comment by ATH1909** [comment]
Ooh, how about a staff of chaos bolt that creates a ghost prompt to take over the body of the creature hit by the staff (even if they already have a client, which would be displaced by the ghost that takes over the body)?

**Comment by tralezab** [comment]
maybe i should directly sort all of the bolts into directly bad, dud, good, and really good (like death for example)

**Comment by ATH1909** [comment]
I like the category idea a lot. What should the weightings of those categories be, though? How often do we want the staff to fire "target is basically donezo" shots?

Sounds like a really good idea, and I wholeheartedly approve of it.

Ooh, don't forget to add a bolt of petrification (for the "target is basically donezo" category, where bolts of death and gibbing would presumably go).

Would making a bolt of baleful polymorph that always turns its target into a relatively harmless creature (a chicken, a mouse, a frog, etc.) just be a matter of making a subtype of normal polymorph bolts that chooses from a different list of polymorph forms, or would it take quite a bit of effort?

EDIT: It seems like it's not that simple, although I'm pretty sure that you can just copy+paste the wabbajack code and just mess with the polymorph form selection system.

Here, I think I've got it:

```
/obj/item/projectile/magic/change
	name = "bolt of change"
	icon_state = "ice_1"
	damage = 0
	damage_type = BURN
	nodamage = 1
        var/baleful = FALSE

/obj/item/projectile/magic/change/baleful
        baleful = TRUE

/obj/item/projectile/magic/change/on_hit(atom/change)
	. = ..()
	if(ismob(change))
		var/mob/M = change
		if(M.anti_magic_check())
			M.visible_message("<span class='warning'>[src] fizzles on contact with [M]!</span>")
			qdel(src)
			return BULLET_ACT_BLOCK
	if(baleful)
                balefulwabbajack(change)
        else
                wabbajack(change)
	qdel(src)

/proc/balefulwabbajack(mob/living/M)
	if(!istype(M) || M.stat == DEAD || M.notransform || (GODMODE & M.status_flags))
		return

	M.notransform = TRUE
	M.mobility_flags = NONE
	M.icon = null
	M.cut_overlays()
	M.invisibility = INVISIBILITY_ABSTRACT

	var/list/contents = M.contents.Copy()

	if(iscyborg(M))
		var/mob/living/silicon/robot/Robot = M
		if(Robot.mmi)
			qdel(Robot.mmi)
		Robot.notify_ai(NEW_BORG)
	else
		for(var/obj/item/W in contents)
			if(!M.dropItemToGround(W))
				qdel(W)

	var/mob/living/new_mob
	
			var/path = pick(/mob/living/simple_animal/parrot,
							/mob/living/simple_animal/pet/dog/corgi,
							/mob/living/simple_animal/crab,
							/mob/living/simple_animal/pet/dog/pug,
							/mob/living/simple_animal/pet/cat,
							/mob/living/simple_animal/mouse,
							/mob/living/simple_animal/chicken,
							/mob/living/simple_animal/cow,
							/mob/living/simple_animal/hostile/lizard,
							/mob/living/simple_animal/pet/fox,
							/mob/living/simple_animal/butterfly,
							/mob/living/simple_animal/pet/cat/cak,
							/mob/living/simple_animal/chick)
			new_mob = new path(M.loc)

	if(!new_mob)
		return
	new_mob.grant_language(/datum/language/common)

	// Some forms can still wear some items
	for(var/obj/item/W in contents)
		new_mob.equip_to_appropriate_slot(W)

	M.log_message("became [new_mob.real_name]", LOG_ATTACK, color="orange")

	new_mob.a_intent = INTENT_HARM

	M.wabbajack_act(new_mob)

	to_chat(new_mob, "<span class='warning'>Your form morphs into that of a [randomize].</span>")

	var/poly_msg = CONFIG_GET(keyed_list/policy)["polymorph"]
	if(poly_msg)
		to_chat(new_mob, poly_msg)

	M.transfer_observers_to(new_mob)

	qdel(M)
	return new_mob
```
I might have gotten the endpoints of that if else setup wrong, but I don't really know how byond decides when to end an if statement (I just assumed that it used how far the code is from the left side of the screen to determine when if statements and else statements end; please correct me if I'm wrong).

Also, if there's a spoiler button available that I can use to cover up that massive block of code, I'd love to be told about it.

**Comment by tralezab** [comment]
it sounds like you're on the cusp of becoming a coder, and you should!

File: code/modules/projectiles/projectile/magic.dm [hunk 9]
```diff
@@ -451,6 +451,117 @@
...
+		var/mob/living/L = target
+		if(L.anti_magic_check())
+			L.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			return BULLET_ACT_BLOCK
+		if(!(L.movement_type & FLYING))
+			return BULLET_ACT_BLOCK
+		L.visible_message("<span class='danger'>[target] is violently ripped out of the air!</span>")
+		L.setMovetype(L.movement_type | FLYING) //will be used with staff of flying for combos often...
+		L.adjustBruteLoss(L.maxHealth/2)//...because the fly magic will throw them back in the air after getting grounded!
+		if(!L.health)
```
Comment (kevinz000) [inline code/modules/projectiles/projectile/magic.dm]: Should be <= SOFTCRIT_THRESHOLD or whatever it is now, pretty sure ! will return false if they're under 0.

Comment (tralezab) [reply code/modules/projectiles/projectile/magic.dm]: thanks for the heads up

**Comment by ExcessiveUseOfCobblestone** [comment]
the desc/cl is pretty barebones for what this PR offers.

**Comment by ATH1909** [comment]
Idea for two "dud" bolts: shrinking and growth bolts. The shrinking bolt would give humanoids the dwarf mutation (and remove their gigantism mutation, if they have it) and shrink other mobs, while the growth bolt would do the opposite.

**Comment by tralezab** [comment]
check the op's edit but i hastily put what i'm doing on a notepad document:

//NEW SET////double the amount of the same-ish level of effects to not change how good the chaos staff is while making bolt of change rarer

bad effect for wizard - 4/4
/obj/item/projectile/magic/resurrection
/obj/item/projectile/magic/change
/obj/item/projectile/magic/antimagic - gives antimagic for 10 seconds
/obj/item/projectile/magic/fetch - brings whatever it hits to you as long as it's not a turf
dud - 4/4
/obj/item/projectile/magic/animate
/obj/item/projectile/magic/door
/obj/item/projectile/magic/nothing - already exists, it's nothing!
/obj/item/projectile/magic/decoration - anomalous crystal's theme warp
slightly good - 6/6
/obj/item/projectile/magic/teleport
/obj/item/projectile/magic/arcane_barrage
/obj/item/projectile/magic/spellblade
/obj/item/projectile/magic/flight - send flying
/obj/item/projectile/magic/necropotence - soul tap
/obj/item/projectile/magic/heat - makes you extremely hot
good - 4/4
/obj/item/projectile/magic/aoe/fireball
/obj/item/projectile/magic/locker
/obj/item/projectile/magic/bounty - heal + spell refresh when target dies
/obj/item/projectile/magic/chill - makes you extremely cold
great - 2/2
/obj/item/projectile/magic/death
/obj/item/projectile/magic/wipe - ghosts immediately, offer to ghosts

**Comment by tralezab** [comment]
i might replace the nothing with bolt of big just because that's a little more chaotic

**Comment by Shadowflame909** [comment]
Aw. No more bolt of magic.

**Comment by PKPenguin321** [comment]
the fetch bolt should throw whatever it hits towards wherever the bolt originated from and beyond

**Comment by tralezab** [comment]
@Shadowflame909 in my opinion one spell if you have no spells is actually fair for how many won't be that useful and the fact that you're already face to face with a wizard that can still kill you in a plethora of ways...

...but chaos carp. it would be much too strong

@PKPenguin321 i can do that, it fucks with the whole stationary bolt of change thing and it keeps the wizard on his feet. thoughts on the new chaos list? bolts that should be in different categories?

**Comment by ATH1909** [comment]
Should the honkmother staff's unique bolt type be added to the list? A bolt of petrification? A bolt of baleful polymorph?

**Comment by ATH1909** [comment]
> Aw. No more bolt of magic.

It looks like the staff of awakening will still be a thing, but chaos magicarps won't shoot bolts of awakening.

**Comment by tralezab** [comment]
No er I'm crunching through my list tonight, haven't committed anything new. No awakening sorry

**Comment by Daxxed** [comment]
As an observe main that ghost bolt seems really neat.
The other changes seem interesting and balanced too, I approve

**Comment by Shadowflame909** [comment]
imo if Subject made a PR removing summon magic, and replaced it with a staff of awakening. That made non-antagonist wizards.


I'd be all for it, and the wizard would have more pain himself, having to deal with targets who know magic standing right in front of him.

**Comment by tralezab** [comment]
here's how the bolt of possession works, i fixed the projectile passing through the victim hit
https://www.youtube.com/watch?v=PXWtXgVbDPc

**Comment by tralezab** [comment]
not sure why diffs did not pass because the rest did fine, but yes should be all good to go

**Comment by ExcessiveUseOfCobblestone** [comment]
no one wants to review lol

**Comment by tralezab** [comment]
day 26! give it up for day 26!

**Comment by ExcessiveUseOfCobblestone** [comment]
@PKPenguin321 if it's not already I don't mind a pickweight system for the SoChaos

File: code/modules/projectiles/guns/magic/staff.dm [hunk 10]
```diff
@@ -105,4 +107,34 @@
...
+
+/obj/item/gun/magic/staff/sapping
+	name = "staff of sapping"
+	desc = "An artefact that spits bolts of sapping magic that can make something sad."
+	fire_sound = 'sound/magic/staff_change.ogg'
+	ammo_type = /obj/item/ammo_casing/magic/sapping
+	icon_state = "staffofsapping"
+	item_state = "staffofsapping"
 
+/obj/item/gun/magic/staff/sapping
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/projectiles/guns/magic/staff.dm]: hello

File: code/modules/projectiles/projectile/magic.dm [hunk 11]
```diff
@@ -451,6 +451,142 @@
...
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			return BULLET_ACT_BLOCK
+		for(var/x in M.get_traumas())//checks to see if the victim is already going through possession
+			if(istype(x, /datum/brain_trauma/special/imaginary_friend/trapped_owner))
+				M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+				return BULLET_ACT_BLOCK
+		to_chat(M, "<span class='warning'>Your mind has been opened to possession!</span>")
+		possession_test(M)
+		return BULLET_ACT_BLOCK
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/projectiles/projectile/magic.dm]: why are you returning block when it has hit at this point?

File: code/modules/projectiles/projectile/magic.dm [hunk 12]
```diff
@@ -451,6 +451,142 @@
...
+	if(LAZYLEN(candidates))
+		var/mob/dead/observer/C = pick(candidates)
+		to_chat(M, "You have been noticed by a ghost, and it has possessed you!")
+		var/oldkey = M.key
+		M.ghostize(0)
+		M.key = C.key
+		trauma.friend.key = oldkey
+		trauma.friend.reset_perspective(null)
+		trauma.friend.Show()
+		trauma.friend_initialized = TRUE
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/projectiles/projectile/magic.dm]: isn't there a transfer mind proc that's safer than just forcing ckey changes?

I'm pretty sure doing that kicks people from the game lol.

File: code/modules/projectiles/projectile/magic.dm (hunk 11 above)
Comment (tralezab) [reply code/modules/projectiles/projectile/magic.dm]: good point

File: code/modules/projectiles/projectile/magic.dm (hunk 12 above)
Comment (tralezab) [reply code/modules/projectiles/projectile/magic.dm]: it's what offer to ghosts does, and what sentience balloons do, do you know where to check for this proc?

Comment (ExcessiveUseOfCobblestone) [reply code/modules/projectiles/projectile/magic.dm]: nvm u ghost first
